DigitalOcean.API/Extensions/RestSharpExtensions.cs

```diff
@@ -43,21 +46,13 @@ public static async Task<T> ExecuteTask<T>(this IRestClient client, IRestRequest
             return response;
         }
 
-        private static Task<IRestResponse> GetResponseContentAsync(IRestClient theClient, IRestRequest theRequest) {
-            var tcs = new TaskCompletionSource<IRestResponse>();
-            theClient.ExecuteAsync(theRequest, response => {
-                tcs.SetResult(response);
-            });
-            return tcs.Task;
+        private static Task<RestResponse> GetResponseContentAsync(IRestClient theClient, RestRequest theRequest) {
+            return theClient.ExecuteAsync(theRequest);
         }
 
-        public static T Deserialize<T>(this IRestResponse response) {
+        public static T Deserialize<T>(this RestResponse response) {
             response.Request.OnBeforeDeserialization(response);
-            var deserialize = new JsonDeserializer {
-                RootElement = response.Request.RootElement,
-                DateFormat = response.Request.DateFormat
-            };
-            return deserialize.Deserialize<T>(response);
+            return JsonDeserializationHelper.DeserializeWithRootElementName<T>(response.Content, response.Request.RootElement);
```

Comment:
We can surely use `JsonSerializerSettings` to set date format. I initially tried to use it but then the issue was with older version getting date format from the `response.Request.DateFormat` property which doesn't exist anymore.
It seems the code was trying to change DO response date format to whatever format was requested by client. Is it possible that we are setting date format in our code where we use DO library? If not, then it will fallback to default date handling.